### PR TITLE
Make job ID accessible to searchkit for joblog

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
JobLog is available in Api4 and in searchkit. But Job ID field is not exposed so you can't filter by job.

Before
----------------------------------------
No Job ID available in searchkit

After
----------------------------------------
Job ID available in searchkit

Technical Details
----------------------------------------

Comments
----------------------------------------
I don't see any sensible reason for this *not* to be a foreign key?

### CODE REVIEW

**Comment by colemanw** [comment]
I don't know why this FK didn't exist in the first place... are we sure this change won't break upgrading any existing sites?

**Comment by colemanw** [comment]
@mattwire this will also need the install sql updated, plus an upgrade step.

**Comment by eileenmcnaughton** [comment]
I think this makes sense but support @colemanw's points.

Also - we should SET NULL on delete I think rather than CASCADE (on upgrade we possibly need to update to NULL in some cases)

**Comment by colemanw** [comment]
Agreed. Setting any invalid FKs to NULL would be a good way to avoid crashing the upgrader.

**Comment by seamuslee001** [comment]
@mattwire can you rebase and change the OnDelete as suggested by Coleman and add the upgrade step as well and then I think we can mere this IMO

File: CRM/Upgrade/Incremental/php/FiveSixty.php [hunk 1]
```diff
@@ -43,4 +44,21 @@ public static function addScheduledReminderSmartySetting(): bool {
     return TRUE;
   }
 
+  public static function addJobLogForeignKey() {
+    // Set job_id = NULL for any that don't have matching jobs (ie. job was deleted).
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FiveSixty.php:48]: I always think a comment in the first line of a function is a pointer it should have a docblock. This should really have a return hint.

Neither are blocking

Comment (mattwire) [reply CRM/Upgrade/Incremental/php/FiveSixty.php:48]: That comment is specific to the first two lines and not the whole function

**Comment by mattwire** [comment]
Added upgrader and ON DELETE SET NULL

File: CRM/Upgrade/Incremental/php/FiveSixty.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FiveSixty.php:48]: @mattwire yeah - I would have put it in a doc block - but as I say, non blocking

**Comment by eileenmcnaughton** [comment]
@mattwire I tested this locally & the FK updates correctly - but the comment isn't updated to remove 'not a FK though, just for logging purposes''

```
CREATE TABLE `civicrm_job_log` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Job log entry ID',
  `domain_id` int(10) unsigned NOT NULL COMMENT 'Which Domain is this scheduled job for',
  `run_time` timestamp NOT NULL DEFAULT current_timestamp() ON UPDATE current_timestamp() COMMENT 'Log entry date',
  `job_id` int(10) unsigned DEFAULT NULL COMMENT 'Pointer to job id - not a FK though, just for logging purposes',
  `name` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL COMMENT 'Title of the job',
  `command` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL COMMENT 'Full path to file containing job script',
  `description` varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL COMMENT 'Title line of log entry',
  `data` longtext COLLATE utf8mb4_unicode_ci DEFAULT NULL COMMENT 'Potential extended data for specific job run (e.g. tracebacks).',
  PRIMARY KEY (`id`),
  KEY `FK_civicrm_job_log_domain_id` (`domain_id`),
  KEY `FK_civicrm_job_log_job_id` (`job_id`),
  CONSTRAINT `FK_civicrm_job_log_domain_id` FOREIGN KEY (`domain_id`) REFERENCES `civicrm_domain` (`id`),
  CONSTRAINT `FK_civicrm_job_log_job_id` FOREIGN KEY (`job_id`) REFERENCES `civicrm_job` (`id`) ON DELETE SET NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci ROW_FORMAT=DYNAMIC
```

**Comment by mattwire** [comment]
@eileenmcnaughton Added an upgrader line to update the comment also

**Comment by eileenmcnaughton** [comment]
@mattwire this failed on something I think it passed on before - different mysql versions?
```
LOGPAT: /home/jenkins/bknix-dfl/build/build-1/.civibuild/private/default/civicrm/ConfigAndLog/CiviCRM.*.log
M02 21 14:34:40  [error] $Fatal Error Details = Array
(
    [callback] => Array
        (
            [0] => CRM_Core_Error
            [1] => exceptionHandler
        )

    [code] => -1
    [message] => DB Error: unknown error
    [mode] => 16
    [debug_info] => UPDATE civicrm_job_log SET job_id = NULL WHERE job_id IN (SELECT job_id from civicrm_job_log cjl
LEFT JOIN civicrm_job cj ON cj.id = cjl.job_id
WHERE cj.id IS NULL) [nativecode=1093 ** You can't specify target table 'civicrm_job_log' for update in FROM clause]
    [type] => DB_Error

```

Maybe

```
UPDATE civicrm_job_log  job_log LEFT JOIN civicrm_job job ON job.id = job_log.id SET job_id = NULL  WHERE job.id IS NULL
```

Seems to work locally

**Comment by eileenmcnaughton** [comment]
@mattwire and now another PR snuck in & made it stale

**Comment by eileenmcnaughton** [comment]
OK - this looks like everything has been addressed.

**Pull request merged**